File: android/src/main/java/com/azifacemobile/i18n/Localization.java
package com.azifacemobile.i18n;

import androidx.annotation.NonNull;

public class Localization {
  public static final String DEFAULT = "en";

  @NonNull
  private String locale;

  public Localization() {
    this.locale = Localization.DEFAULT;
  }

  @NonNull
  public String getLocale() {
    return this.locale;
  }

  public void setLocale(@NonNull String locale) {
    String temp = locale.toLowerCase();

    switch (temp) {
      case "af":
      case "ar":
      case "de":
      case "el":
      case "es":
      case "fr":
      case "ja":
      case "kk":
      case "nb":
      case "ru":
      case "vi":
      case "zh":
        this.locale = temp;
        break;
      case "pt-br":
        this.locale = "pt";
        break;
      default:
        this.locale = "en";
        break;
    }
  }
}

File: ios/I18n/Localization.swift
import Foundation

public class Localization {
  private var locale: String = "default"
  public static let DEFAULT = "en"

  public init() {
    self.locale = Localization.DEFAULT
  }

  public func getLocale() -> String {
    return self.locale
  }

  public func setLocale(locale: String) {
    let temp = locale.lowercased()

    switch (temp) {
    case "af", "ar", "de", "el", "es", "fr", "ja", "kk", "nb", "ru", "vi", "zh":
      self.locale = temp
      break
    case "pt-br":
      self.locale = "pt-BR"
      break
    default:
      self.locale = "en"
      break
    }
  }
}
